### ISSUE TITLE
References: tree and peek should pass focus to editor on Enter

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#104374]
Currently the peek references and the references tree on Enter do not pass focus to the editor.
This is not aligned with how all other trees behave (explorer, problems, search, git)

Also for accessibility reasons it would be nice if all trees behave the same.
That `space` leaves focus on the tree and executes the action. And that `enter` moves focus.

Let me know what you think

**Comment by jrieken** [comment]
> Currently the peek references and the references tree on Enter do not pass focus to the editor.

Two issues in one? So, for peek this doesn't make sense as enter is the "close peek and reveal reference in outer editor" gesture. Are you saying that instead the inner editor should be focused?

**Comment by isidorn** [comment]
For me: peek on `enter` does not close the close the peek, but simply moves it - gif attached. On `space` it does not do anything.
I suggest: `enter` - close the peek and reveal reference in the outer editor and pass focus to outer editor.
`space` - what currently enter does

![recording (3)](https://user-images.githubusercontent.com/1926584/89794763-cabfdb00-db27-11ea-94f2-f738632392d4.gif)

**Comment by jrieken** [comment]
Using latest insiders I don't see what you are saying. I see enter opening a reference and space doing nothing

[Aug-10-2020 16-40-32.gif.zip](https://github.com/microsoft/vscode/files/5051470/Aug-10-2020.16-40-32.gif.zip)

**Comment by isidorn** [comment]
Oh now I see. If I do context menu > Go To References it beahves as you say.
If I do context menu > Peek Refereneces it does not. Which I guess is the difference between these two experiences. Which makes sense, I was just unaware of it.

So you can just treat this issue as a suggestion to change the Find All References tree view

**Comment by jrieken** [comment]
> Which I guess is the difference between these two experiences. Which makes sense, I was just unaware of it.

Yeah, that the difference. WHich is quite subtle..

> ust treat this issue as a suggestion to change the Find All References tree view

You be able to find a duplicate for that ;-) I currently lacks API as the tree view doesn't let know how an item was triggered...

**Comment by Neurrone** [comment]
Hm, wondering if this issue should be reopened?

If the references tree does not move focus to the editor upon pressing enter, that sounds similar to an existing issue, and it would be nice if trees all behaved predictably.
